### DISCUSSION TITLE
基盤にmicroproを挿し込むとPC側で認識できなくなる

### DISCUSSION THREAD

**Discussion by ancoromochi100** [Yowkees/keyball#518] [Q&A]
質問失礼致します。
現在keyball61の作成をしています。

【現状】
基盤に挿し込む前にusb接続をするとキーボードとして認識できるのですが、基盤に取り付けると認識できなくなってしまいます。

【原因】
可能性としてはOLEDのソケットの半田付けに失敗してしまったのですが、原因となりうるでしょうか…？

お手数をおかけしてしまい恐縮ですが、お助けいただけると幸いです。
よろしくお願いいたします。

![IMG_7108](https://github.com/Yowkees/keyball/assets/99373491/c2261770-b6dc-4090-8061-d00e7d3dfd50)

**Comment by koron** [comment]
(issue から discussion へ変更しました)

> OLEDのソケットの半田付けに失敗してしまったのですが、原因となりうるでしょうか…？

OLEDソケットの可能性もゼロではありませんが、先にLEDを疑ってみてください。

Keyballに限らず、Pro Microが単体では動くのに、
自作キーボードの基板に指すと機能しなくなってしまう場合、
基板側でどこか電源とアース(GND)がショートしていることがよくあります。

そうなっていないかは、Pro Microを基板に挿入しUSBも繋ぎ電源を入れた状態で、
テスターを使ってVCCとGNDの間の電圧を測定することでわかります。
Pro Microが機能するためにはVCCのほうがGNDよりも
およそ3.3V、できれば4.6V以上電圧が高い状態である必要があります。

起動しない場合は電圧が3Vを下回るか0Vになっていることが多いです。

それが発生する原因として、私が体験したことがあるのは
LED SK6812のハンダがショートしていたり、
裏表反対に付けていたり左右を反対に付けていたり…

なのでまずはLEDをすべてチェックしてみることをオススメします。
